### ISSUE TITLE
Are images and labels shuffled through the dataloader

### ISSUE THREAD

**Issue by JSGrondin** [ultralytics/yolov5#761]
## ❔Question
When looking at the function create_dataloader in dataset.py, I see that the dataloader doesn't include the argument shuffle=True, which means the data is not shuffled after each epoch. It is not clear to me whether the data is at least shuffled once at the beginning of training when shuffle=False or if the data is simply loaded in the alphanumerical order of the image/label file names? Could anyone clarify this please?

**Comment by glenn-jocher** [comment]
@JSGrondin data is shuffled for training and sorted by aspect ratio for batched rectangular inference during validation.

**Comment by JSGrondin** [comment]
Many thanks!

**Comment by sramakrishnan247** [comment]
@glenn-jocher 
> @JSGrondin data is shuffled for training and sorted by aspect ratio for batched rectangular inference during validation.
Can you please tell me where exactly this is happening?

**Comment by glenn-jocher** [comment]
@JSGrondin in the dataloader:
https://github.com/ultralytics/yolov5/blob/9d873077841434d1c6cbd1c4248ca2252820d3ba/utils/datasets.py#L341

**Comment by glenn-jocher** [comment]
@JSGrondin @sramakrishnan247 good news 😃! Your original issue may now be fixed ✅ in PR #5623 by @werner-duvaud. This PR turns on shuffling in the YOLOv5 training DataLoader by default, which was missing until now. This works for all training formats: CPU, Single-GPU, Multi-GPU DDP.

```python
train_loader, dataset = create_dataloader(train_path, imgsz, batch_size // WORLD_SIZE, gs, single_cls,
                                          hyp=hyp, augment=True, cache=opt.cache, rect=opt.rect, rank=LOCAL_RANK,
                                          workers=workers, image_weights=opt.image_weights, quad=opt.quad,
                                          prefix=colorstr('train: '), shuffle=True)  # <--- NEW
```

I evaluated this PR against master on VOC finetuning for 50 epochs, and the results show a slight improvement in most metrics and losses, particularly in objectness loss and mAP@0.5, perhaps indicating that the shuffle addition may help delay overtraining.

https://wandb.ai/glenn-jocher/VOC
<img width="979" alt="Screenshot 2021-11-13 at 13 03 26" src="https://user-images.githubusercontent.com/26833433/141643075-70b06810-4727-4f5c-b03a-f68c34fe6934.png">

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by kevinconka** [comment]
> @JSGrondin data is shuffled for training and sorted by aspect ratio for batched rectangular inference during validation.

I would like to shuffle my validation data so that in clearML I can see 'shuffled' samples. I have tried the following in `val.py` but did not work:
<img width="627" alt="image" src="https://user-images.githubusercontent.com/35779409/208139519-4973c90e-5438-46fc-9ce9-2f1cbbe6e711.png">

Is there something I am missing?

**Comment by glenn-jocher** [comment]
@kevinconka shuffling in validation set is irrelevant for metrics purposes.
